susemanager-utils/susemanager-sls/src/modules/mgractionchains.py

```diff
@@ -57,22 +57,39 @@ def _get_ac_storage_filenamepath():
 
     return os.path.join(minion_d_dir, '_mgractionchains.conf')
 
+def check_reboot_required(target_sls):
+    sls_file_on_minion = __salt__['cp.cache_file']('{0}{1}.sls'.format('salt://actionchains/', target_sls.replace('actionchains.','')))
+    yaml_file = _read_sls_file(sls_file_on_minion)
+    list_param = yaml_file['schedule_next_chunk']['mgrcompat.module_run']
+
+    for dic in list_param:
+        if 'reboot_required' in dic.keys() :
+            return dic.values()
```

Comment:
I think this might not be exactly what we need here. `dict.values()` returns a memory-view of values, but the only caller just evaluates the truthiness of `check_reboot_required()`. This check will always be `True` as long as `reboot_required` is in the `dict`, no matter what the value is:

```python
>>> def check():
...     return {"reboot_required": False}.values()
... 
>>> required = check()
>>> bool(required)
True
```

So what this is doing is basically the same as 
```python
return 'reboot_required' in dic
```
There are two options now: change the calling or the called function:

```python
reboot_required = next(iter(check_reboot_required(target_sls)))
```
This approach has a big downside: `check_reboot_required` does not always return something that can be turned into an `iterator`. If `reboot_required` is not a parameter, the return will be `False` and
```python
>>> iter(False)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'bool' object is not iterable
```
There would be another problem, and that is that `next()` is shadowed in this module. 

That brings us to changing the return value of this function. Based on the function name, I'd only expect a boolean anyway, so it's a good idea to change this loop.

```python
    for dic in list_param:
        if 'reboot_required' in dic: # or dict.keys(), they are equivalent here
            return dic["reboot_required"]
```

Reply:
thanks for the explanation! I'll fix it